File: Binary-Search/0153-find-minimum-in-rotated-sorted-array.py
# Time  : O(logn)
# Space : O(1)

class Solution:
  def findMin(self, A: List[int]) -> int:  # pivot
    lo, hi = 0, len(A) - 1

    while lo < hi:
      mid = lo + (hi - lo) // 2
      if A[mid] > A[hi]:
        lo = mid + 1
      else:
        hi = mid

    return A[lo]

File: Binary-Search/0153-find-minimum-in-rotated-sorted-array.cpp
// Time  : O(logn)
// Space : O(1)
class Solution {
public:
  int findMin(vector<int>& A) {
    int lo = 0;
    int hi = A.size()-1;
    while (lo < hi) {
      int mid = (lo + hi)>>1;
      if (A[mid] > A[hi]) lo = mid+1;
      else hi = mid;
    }
    return A[lo];
  }
};